src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,91 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
+import './NewMovie.scss';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (newMovie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+
+  const newMovieCard = {
+    title,
+    description,
+    imgUrl,
+    imdbUrl,
+    imdbId,
+  };
+
+  const clearForm = () => {
+    setCount(prev => prev + 1);
+    setTitle('');
+    setDescription('');
+    setImgUrl('');
+    setImdbUrl('');
+    setImdbId('');
+  };
+
+  const canSubmit = title.trim() !== ''
+    && imgUrl.trim() !== ''
+    && imdbUrl.trim() !== ''
+    && imdbId.trim() !== '';
+
```

Comment:
```suggestion
  const canSubmit = Boolean(
    title.trim()
    && imgUrl.trim()
    && imdbUrl.trim()
    && imdbId.trim(),
```

and u can trim inside textFields

Reply:
✔️